### PR TITLE
Nina vectorization spd

### PR DESCRIPTION
**Pull request by ninamiolane**
Vectorization of module on symmetric positive definite matrices

### CODE REVIEW

File: geomstats/riemannian_metric.py [hunk 1]
```diff
@@ -193,9 +193,36 @@ def geodesic(self, initial_point, initial_tangent_vec):
...
+                            'non-compatible shapes: {} and {}.'.format(
+                                                  initial_point.shape,
+                                                  initial_tangent_vec.shape))
+
+            n_times_t, _ = t.shape
+            tangent_vec_shape = new_initial_tangent_vec.shape[1:]
+            tangent_vecs = np.zeros((n_times_t,) + tangent_vec_shape)
+            for i in range(n_times_t):
+                tangent_vecs[i] = t[i] * new_initial_tangent_vec
+            # tangent_vecs = np.outer(t, new_initial_tangent_vec)
```
Comment (johmathe) [inline geomstats/riemannian_metric.py]: Leftover comment

File: geomstats/spd_matrices_space.py [hunk 2]
```diff
@@ -18,12 +18,25 @@
...
+        mat = np.expand_dims(mat, axis=0)
+    assert mat.ndim == 3
+    n_mats, _, _ = mat.shape
+
+    mask = np.zeros(n_mats, dtype=bool)
+    for i in range(n_mats):
+        mask[i] = np.allclose(mat[i], np.transpose(mat[i]),
+                              atol=tolerance)
+
+    return mask
```
Comment (johmathe) [inline geomstats/spd_matrices_space.py]: maybe mask doesn’t feel like the best variable name here. Also can’t you get rid of the for loop here by doing the allclose along the first axis or something ?

File: geomstats/spd_matrices_space.py [hunk 3]
```diff
@@ -107,42 +143,70 @@ def vector_to_matrix(self, vector):
         """
         Convert a vector into a symmetric matrix.
         """
+        if vector.ndim == 1:
+            vector = np.expand_dims(vector, axis=0)
+        assert vector.ndim == 2
         # TODO(nina): why factor np.sqrt(2)
```
Comment (johmathe) [inline geomstats/spd_matrices_space.py]: This todo is not super clear

File: geomstats/spd_matrices_space.py [hunk 4]
```diff
@@ -153,17 +217,35 @@ def exp(self, tangent_vec, base_point):
 
         This gives a symmetric positive definite matrix.
         """
-        sqrt_base_point = scipy.linalg.sqrtm(base_point)
+        if tangent_vec.ndim == 2:
```
Comment (johmathe) [inline geomstats/spd_matrices_space.py]: Seems that you’re using this patter quite often in your code. Want to refactor this?

File: geomstats/spd_matrices_space.py (hunk 2 above)
Comment (ninamiolane) [reply geomstats/spd_matrices_space.py]: I changed mask to is_symmetric_bool. 

Yes I wanted to get rid of the for loop, but there's no axis parameter in allclose and I didn't want to hack something.

File: geomstats/spd_matrices_space.py (hunk 4 above)
Comment (ninamiolane) [reply geomstats/spd_matrices_space.py]: Good point, thank you. I will add a TODO for now.

**Pull request merged**